src/Manager/Environment/effects/removeEnvironment.ts

```diff
@@ -23,27 +23,42 @@ import {
 } from '../environmentReducer';
 import { removeDir } from './removeDir';
 
-const removeLegacyEnvironment = (dispatch: Dispatch, toolchainDir: string) =>
-    removeDir(dispatch, path.dirname(toolchainDir));
+const removeLegacyEnvironment = (toolchainDir: string) =>
+    removeDir(path.dirname(toolchainDir));
 
-const removeNrfutilEnvironment = (dispatch: Dispatch, version: string) =>
-    Promise.all([
-        removeToolchain(version),
-        removeDir(dispatch, sdkPath(version)),
-    ]);
+const removeNrfutilEnvironment = (version: string) => {
+    removeDir(sdkPath(version));
+    try {
+        removeToolchain(version);
+    } catch (err) {
+        throw new Error(
+            `Failed to remove ${sdkPath(version)}, ${err}. ` +
+                `Please remove the installation for ${version} manually as it is broken.`
+        );
+    }
+};
 
 export const removeEnvironment =
-    (environment: Environment) => async (dispatch: Dispatch) => {
+    (environment: Environment) => (dispatch: Dispatch) => {
         const { toolchainDir, version } = environment;
         logger.info(`Removing ${version} at ${toolchainDir}`);
         usageData.sendUsageData(EventAction.REMOVE_TOOLCHAIN, `${version}`);
 
         dispatch(startRemoving(version));
 
-        if (isLegacyEnvironment(version)) {
-            await removeLegacyEnvironment(dispatch, toolchainDir);
-        } else {
-            await removeNrfutilEnvironment(dispatch, version);
+        try {
+            if (isLegacyEnvironment(version)) {
+                removeLegacyEnvironment(toolchainDir);
+            } else {
+                removeNrfutilEnvironment(version);
+            }
+        } catch (err) {
+            dispatch(
+                ErrorDialogActions.showDialog(`${(err as Error).message}`)
+            );
+            usageData.sendErrorReport((err as Error).message);
```

Comment:
It is fine, to use `err as Error` here (as we also talked about). 

If you are interested in another solution: You could also define own subclasses of `Error`. Either special-purpose subclasses or a general purpose like this:
```ts
class PlainMessageError extends Error {
    toString = () => this.message;
}
```

Then you can throw `PlainMessageError` instead of `Error` in `removeDir` and `removeNrfutilEnvironment`. And here you can replace `(err as Error).message` with `describeError(err`